packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -4558,22 +4541,12 @@ class _InputDecoratorDefaultsM2 extends InputDecorationTheme {
 
   @override
   Color? get fillColor => MaterialStateColor.resolveWith((Set<MaterialState> states) {
-    if (states.contains(MaterialState.disabled)) {
-      // dark theme: 5% white
-      // light theme: 2% black
-      switch (Theme.of(context).brightness) {
-        case Brightness.dark:
-          return const Color(0x0DFFFFFF);
-        case Brightness.light:
-          return const Color(0x05000000) ;
-      }
-    }
-    // dark theme: 10% white
-    // light theme: 4% black
-    switch (Theme.of(context).brightness) {
-      case Brightness.dark: return const Color(0x1AFFFFFF);
-      case Brightness.light:return const Color(0x0A000000) ;
-    }
+    return switch ((Theme.of(context).brightness, states.contains(MaterialState.disabled))) {
+      (Brightness.dark,  true ) => const Color(0x0DFFFFFF), //  5% white
+      (Brightness.dark,  false) => const Color(0x1AFFFFFF), // 10% white
+      (Brightness.light, true ) => const Color(0x05000000), //  2% black
+      (Brightness.light, false) => const Color(0x0A000000), //  4% black
```

Comment:
Please avoid spaces in the pattern matching. Instead of `Brightness.dark, true )

```suggestion
      (Brightness.dark, true)   => const Color(0x0DFFFFFF), //  5% white
      (Brightness.dark, false)  => const Color(0x1AFFFFFF), // 10% white
      (Brightness.light, true)  => const Color(0x05000000), //  2% black
      (Brightness.light, false) => const Color(0x0A000000), //  4% black
```